src/services/tx/tx-sender/create.ts

```diff
@@ -1,121 +1,56 @@
-import { getTransactionDetails, Operation, postSafeGasEstimation } from '@safe-global/safe-gateway-typescript-sdk'
+import type { TransactionDetails } from '@safe-global/safe-gateway-typescript-sdk'
+import { getTransactionDetails } from '@safe-global/safe-gateway-typescript-sdk'
 import type { AddOwnerTxParams, RemoveOwnerTxParams, SwapOwnerTxParams } from '@safe-global/safe-core-sdk'
 import type { MetaTransactionData, SafeTransaction, SafeTransactionDataPartial } from '@safe-global/safe-core-sdk-types'
-import { EMPTY_DATA } from '@safe-global/safe-core-sdk/dist/src/utils/constants'
 import extractTxInfo from '../extractTxInfo'
 import { getAndValidateSafeSDK } from './sdk'
-import type Safe from '@safe-global/safe-core-sdk'
-
-const estimateSafeTxGas = async (
-  chainId: string,
-  safeAddress: string,
-  txParams: MetaTransactionData,
-): Promise<SafeTransactionEstimation> => {
-  return postSafeGasEstimation(chainId, safeAddress, {
-    to: txParams.to,
-    value: txParams.value,
-    data: txParams.data,
-    operation: (txParams.operation as unknown as Operation) || Operation.CALL,
-  })
-}
-
-const getRecommendedTxParams = async (
-  txParams: SafeTransactionDataPartial,
-): Promise<{ nonce: number; safeTxGas: number } | undefined> => {
-  const safeSDK = getAndValidateSafeSDK()
-  const chainId = await safeSDK.getChainId()
-  const safeAddress = safeSDK.getAddress()
-  const contractVersion = await safeSDK.getContractVersion()
-  const isSafeTxGasRequired = isLegacyVersion(contractVersion)
-
-  let estimation: SafeTransactionEstimation | undefined
-
-  try {
-    estimation = await estimateSafeTxGas(String(chainId), safeAddress, txParams)
-  } catch (e) {
-    try {
-      // If the initial transaction data causes the estimation to fail,
-      // we retry the request with a cancellation transaction to get the
-      // recommendedNonce, even if the original transaction will likely fail
-      const cancellationTxParams = { ...txParams, data: EMPTY_DATA, to: safeAddress, value: '0' }
-      estimation = await estimateSafeTxGas(String(chainId), safeAddress, cancellationTxParams)
-    } catch (e) {
-      logError(Errors._616, (e as Error).message)
-      return
-    }
-  }
-
-  return {
-    nonce: estimation.recommendedNonce,
-    safeTxGas: isSafeTxGasRequired ? Number(estimation.safeTxGas) : 0,
-  }
-}
 
 /**
  * Create a transaction from raw params
  */
 export const createTx = async (txParams: SafeTransactionDataPartial, nonce?: number): Promise<SafeTransaction> => {
+  if (nonce !== undefined) txParams = { ...txParams, nonce }
   const safeSDK = getAndValidateSafeSDK()
-
-  // If the nonce is not provided, we get the recommended one
-  if (nonce === undefined) {
-    const recParams = (await getRecommendedTxParams(txParams)) || {}
-    txParams = { ...txParams, ...recParams }
-  } else {
-    // Otherwise, we use the provided one
-    txParams = { ...txParams, nonce }
-  }
-
```

Comment:
I just realized that we don't estimate the nonce anymore for the initial tx that is passed to `SignOrExecuteForm`. Thats probably the reason why the execute checkbox is showing initially. I suggest we restore this block.

Reply:
It also applies to the other `create` calls. We might need the `withRecommendedNonce` helper after all.